### PR TITLE
Waters

### PR DESCRIPTION
**Pull request by remi-dupre**
Include some `water` elements in elasticsearch for the geocoder.

Also adds a builder pattern in `pg_poi_query.rs` to construct the SQL query used to fetch all POIs, this will be useful for https://github.com/QwantResearch/fafnir/pull/51.

### CODE REVIEW

**Comment by remi-dupre** [comment]
@amatissart I just added the least amount of tests I wanted before really asking for a review.

I'll focus on the bad error raising of  [this branch](https://github.com/QwantResearch/fafnir/tree/waters-failing) later as I started being very inefficient when working on it, I bet I just need to work on something else for a while ^^

File: src/lib.rs [hunk 1]
```diff
@@ -96,70 +97,45 @@ pub fn load_and_index_pois(
...
+        )
+        .with_table(
+            TableQuery::new("osm_water_lakeline")
+                .override_class("water")
+                .override_subclass("lake"),
+        )
+        .with_table(
+            TableQuery::new("osm_water_point")
+                .override_class("water")
+                .override_subclass("river"),
```
Comment (amatissart) [inline src/lib.rs:123]: Is there a more specific value we could use, instead of hardcoding "river" ?

Comment (remi-dupre) [reply src/lib.rs:123]: And this default is actually quite bad, I'll change it to `water` too if we don't find any better.

We could attempt to compute a subclass like we do for POIs for the waters with an amenity or landuse tag.

File: src/pg_poi_query.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            ",
+            table = self.table,
+            id_column = self.id_column,
+            class = self
+                .override_class
+                .as_ref()
+                .map_or_else(|| "class".to_string(), |name| format!("{} AS class", name)),
+            mapping_key = self.override_class.as_ref().map_or_else(
+                || "mapping_key".to_string(),
+                |name| format!("{} AS class", name)
```
Comment (amatissart) [inline src/pg_poi_query.rs:121]: typo here ?
```suggestion
                |name| format!("{} AS mapping_key", name)
```

File: src/pg_poi_query.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            ),
+            subclass = self.override_subclass.as_ref().map_or_else(
+                || "subclass".to_string(),
+                |name| format!("{} AS subclass", name)
+            ),
+            geometry_point = "ST_Transform(ST_Centroid(geometry), 4326)",
```
Comment (amatissart) [inline src/pg_poi_query.rs:127]: [`ST_Centroid`](https://postgis.net/docs/ST_Centroid.html) may return point outside of the geometry. 
[`ST_PointOnSurface`](https://postgis.net/docs/ST_PointOnSurface.html) is probably preferable.

```suggestion
            geometry_point = "ST_Transform(ST_PointOnSurface(geometry), 4326)",
```

File: src/lib.rs (hunk 1 above)
Comment (amatissart) [reply src/lib.rs]: yes using 'water' both for class and subclass is probably good enough for now, as the value of the `water=` tag does not seem to be imported in the database.

**Review by amatissart — APPROVED** [review]
Do you know why the format of Cargo.lock changed ? Is it related to your version of `cargo` ?

**Comment by remi-dupre** [comment]
> Do you know why the format of Cargo.lock changed ? Is it related to your version of `cargo` ?

I'm not sure why this append but cargo allowed me to change this back

**Pull request merged**